src/System.Reflection/tests/MethodInfo/MethodInfo_ToStringTests.cs

```diff
@@ -11,124 +11,49 @@ namespace System.Reflection.Tests
-
-            //case 4
-            VerifyMethodSignature(type, "CompareExchange", "Int32 CompareExchange(Int32 ByRef, Int32, Int32)");
-        }
-
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature4()
-        {
-            VerifyMethodSignature("DummyMethod2", "Void DummyMethod2()");
-        }
-
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature5()
-        {
-            VerifyMethodSignature(typeof(MethodInfoToStringSample), "Method1", "System.String Method1(System.DateTime)");
-        }
-
-
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature6()
-        {
-            VerifyMethodSignature(typeof(MethodInfoToStringSample), "Method2", "System.String Method2[T,S](System.String, T, S)");
-        }
-
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature7()
+        //Verify Method Signatures ref parameters
+        [Theory]
+        [MemberData(nameof(TestMethodData))]
+        public static void TestMethodSignature(Type type, string methodName, string methodSign)
         {
-            VerifyMethodSignature(typeof(MethodInfoToStringSampleG<>), "Method1", "T Method1(T)");
-        }
+            MethodInfo mi = getMethod(type, methodName);
 
+            Assert.NotNull(mi);
 
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature8()
-        {
-            VerifyMethodSignature(typeof(MethodInfoToStringSampleG<>), "Method2", "T Method2[S](S, T, System.String)");
-        }
+            Assert.True(mi.Name.Equals(methodName, StringComparison.CurrentCultureIgnoreCase));
 
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature9()
-        {
-            VerifyMethodSignature(typeof(MethodInfoToStringSampleG<string>), "Method1", "System.String Method1(System.String)");
+            Assert.True(mi.ToString().Equals(methodSign, StringComparison.CurrentCultureIgnoreCase));
         }
-
+       
         //Verify Method Signatures
         [Fact]
-        public static void TestMethodSignature10()
-        {
-            VerifyMethodSignature(typeof(MethodInfoToStringSampleG<string>), "Method2", "System.String Method2[S](S, System.String, System.String)");
-        }
-
-
-        //Verify Method Signatures
-        [Fact]
-        public static void TestMethodSignature11()
+        public static void TestMethodSignature()
         {
             //Make generic method
             MethodInfo gmi = getMethod(typeof(MethodInfoToStringSampleG<string>), "Method2").MakeGenericMethod(new Type[] { typeof(DateTime) });
-            String sign = "System.String Method2[DateTime](System.DateTime, System.String, System.String)";
+            string sign = "System.String Method2[DateTime](System.DateTime, System.String, System.String)";
```

Comment:
Nit: since ```sign``` is a word that's often seen in tests and has a completely different meaning, it might be good to pick a different variable name, like ```signature```.